vllm/sampling_params.py

```diff
@@ -246,10 +253,12 @@ def clone(self) -> "SamplingParams":
         See https://github.com/vllm-project/vllm/issues/3087
         """
 
-        logit_processor_refs = None if self.logits_processors is None else {
+        logit_processor_refs = (None if self.logits_processors is None else {
             id(lp): lp
             for lp in self.logits_processors
-        }
+            if (not hasattr(lp[0] if isinstance(lp, Sequence) else lp, "fsm")
+                if lp is not None else True)
+        })
```

Comment:
This code is becoming hard to understand, can you rewrite it?

Reply:
Sure